### ISSUE TITLE
ACME-Client: Cannot find DNS API hook for: dns_nsupdate

### ISSUE THREAD

**Issue by Phreeman33** [opnsense/plugins#4264]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

**Describe the bug**

Similar to [this issue](https://github.com/opnsense/plugins/issues/4218).
When using ACME-Client to configure the API with RFC2136, the request or update of the certificate fails, and no specific reason is shown in the log files. Then enter the command from the log into the console, it displays "Cannot find DNS API hook for: dns_nsupdate."

**Relevant log files**

Logs from the file:
[Sat Sep 28 21:41:53 CST 2024] See: https://github.com/acmesh-official/acme.sh/wiki/How-to-debug-acme.sh
[Sat Sep 28 21:41:53 CST 2024] Please add '--debug' or '--log' to see more information.
[Sat Sep 28 21:41:53 CST 2024] Error adding TXT record to domain: _acme-challenge.mydomain.tld
[Sat Sep 28 21:41:53 CST 2024] error updating domain
[Sat Sep 28 21:41:53 CST 2024] adding _acme-challenge.mydomain.tld. 60 in txt "FaBVqRQfqohelqAfwu9kLjDbvhJVm_bboS7x-bsQu7E"
[Sat Sep 28 21:41:53 CST 2024] Adding TXT value: FaBVqRQfqohelqAfwu9kLjDbvhJVm_bboS7x-bsQu7E for domain: _acme-challenge.mydomain.tld
[Sat Sep 28 21:41:53 CST 2024] Getting webroot for domain='\*.mydomain.tld'
[Sat Sep 28 21:41:53 CST 2024] Getting webroot for domain='mydomain.tld'
[Sat Sep 28 21:41:50 CST 2024] Multi domain='DNS:mydomain.tld,DNS:\*.mydomain.tld'
[Sat Sep 28 21:41:50 CST 2024] Using CA: https://acme-staging-v02.api.letsencrypt.org/directory

Logs from console:
[Sat Sep 28 21:42:49 CST 2024] Using CA: https://acme-staging-v02.api.letsencrypt.org/directory
[Sat Sep 28 21:42:49 CST 2024] Multi domain='DNS:mydomain.tld,DNS:\*.mydomain.tld'
[Sat Sep 28 21:42:52 CST 2024] Getting webroot for domain='mydomain.tld'
[Sat Sep 28 21:42:52 CST 2024] Getting webroot for domain='\*.mydomain.tld'
[Sat Sep 28 21:42:52 CST 2024] Cannot find DNS API hook for: dns_nsupdate
[Sat Sep 28 21:42:52 CST 2024] You need to add the TXT record manually.
[Sat Sep 28 21:42:52 CST 2024] Add the following TXT record:
[Sat Sep 28 21:42:52 CST 2024] Domain: '_acme-challenge.mydomain.tld'
[Sat Sep 28 21:42:52 CST 2024] TXT value: 'EO45x0F4io_CouFe3oj3-91EpNF9VyhZtOE1y_tOAu4'
[Sat Sep 28 21:42:52 CST 2024] Please make sure to prepend '_acme-challenge.' to your domain
[Sat Sep 28 21:42:52 CST 2024] so that the resulting subdomain is: _acme-challenge.mydomain.tld
[Sat Sep 28 21:42:52 CST 2024] Cannot find DNS API hook for: dns_nsupdate
[Sat Sep 28 21:42:52 CST 2024] You need to add the TXT record manually.
[Sat Sep 28 21:42:52 CST 2024] Add the following TXT record:
[Sat Sep 28 21:42:52 CST 2024] Domain: '_acme-challenge.mydomain.tld'
[Sat Sep 28 21:42:52 CST 2024] TXT value: 'ZrsyQVYev59HxkVVgivxAFCuA0a8UT70meUdwZ462AI'
[Sat Sep 28 21:42:52 CST 2024] Please make sure to prepend '_acme-challenge.' to your domain
[Sat Sep 28 21:42:52 CST 2024] so that the resulting subdomain is: _acme-challenge.mydomain.tld
[Sat Sep 28 21:42:52 CST 2024] Please add the TXT records to the domains, and re-run with --renew.
[Sat Sep 28 21:42:52 CST 2024] Please add '--debug' or '--log' to see more information.
[Sat Sep 28 21:42:52 CST 2024] See: https://github.com/acmesh-official/acme.sh/wiki/How-to-debug-acme.sh

**Environment**

OPNsense 24.7.5-amd64
Intel® Core™ i5-3340 3.1Ghz Quad Core
Network Intel® I350-T4

**Comment by Phreeman33** [comment]
After fresh installed 24.7, the issue still existed. Also, encountered the blank log issue as [this](https://github.com/opnsense/plugins/issues/4217).

**Comment by jlrgraham** [comment]
For me downgrading to `acme.sh-3.0.7_2` resolves the issue.

**Comment by fraenki** [comment]
> Cannot find dns api hook for:

Although I have not been able to reproduce this, the following patch may fix this issue:

```
opnsense-patch -c plugins d1ef1f93ce318926c14e14291a50c09ac6421f15
```

Please apply the patch and report whether or not this fixes the infamous `
Cannot find dns api hook for:` error message.

If this does _not_ fix this error, please use the Acme Client GUI to change `Log Level` to `debug 3` and try again. Afterwards please provide the resulting ACME Log.

**Comment by Phreeman33** [comment]
> > Cannot find dns api hook for:
> 
> Although I have not been able to reproduce this, the following patch may fix this issue:
> 
> ```
> opnsense-patch -c plugins d1ef1f93ce318926c14e14291a50c09ac6421f15
> ```
> 
> Please apply the patch and report whether or not this fixes the infamous ` Cannot find dns api hook for:` error message.
> 
> If this does _not_ fix this error, please use the Acme Client GUI to change `Log Level` to `debug 3` and try again. Afterwards please provide the resulting ACME Log.

Works fine for me, but still got an error from console: `could not open '': file not found"`.
Seemed this [issue](https://github.com/acmesh-official/acme.sh/pull/5285) was already fixed on upstream realese [3.0.9](https://github.com/acmesh-official/acme.sh/releases/tag/3.0.9), I temporary manual applied it and looked good.

**Comment by fraenki** [comment]
> Worked fine for me

Thanks for testing!

> Seemed this issues was already fixed on upstream realese 3.0.9

Yes, this was my PR :grin: – will be available in the next OPNsense minor release.
